model/publishing/delivery/tasks/RemoteTaskStatusSynchroniser.php

```diff
@@ -21,40 +28,177 @@ class RemoteTaskStatusSynchroniser implements Action,ServiceLocatorAwareInterfac
 
     private $status;
 
+    private $remoteTaskId;
+    private $remoteEnvironmentId;
+
+    private $remoteDeliveryId = null;
+
     public function getRemoteStatus()
     {
         return $this->status;
     }
 
     /**
      * @param $params
-     * @return \common_report_Report
+     * @return Report
+     * @throws \common_exception_Error
      * @throws \common_exception_MissingParameter
      */
     public function __invoke($params)
     {
-        if (count($params) != 2) {
+        if (count($params) != 4) {
             throw new \common_exception_MissingParameter();
         }
-        $taskId = array_shift($params);
-        $envId = array_shift($params);
+        list ($remoteTaskId, $envId, $deliveryUri, $testUri) = $params;
+        $this->remoteTaskId = $remoteTaskId;
+        $this->remoteEnvironmentId = $envId;
+
+        try {
+            $this->status = $this->fetchRemoteTaskStatus();
```

Comment:
I see, eventually you ended up using 2 calls for getting status and report data. 
It's very bizarre that `TaskQueue/get` doesn't return the same status. 

It works like that, fine by me!

Reply:
AbstractWorker relies on status returned by `TaskQueue/getStatus` endpoint. I could use `get` and map status to what is expected by AbstractWorker, but decided to have two calls in the end. Second call is made only when task is completed.